application/src/main/java/org/togetherjava/tjbot/commands/moderation/AuditCommand.java

```diff
@@ -139,26 +138,58 @@ private boolean handleChecks(@NotNull Member bot, @NotNull Member author,
         return ModerationUtils.handleHasAuthorRole(ACTION_VERB, hasRequiredRole, author, event);
     }
 
-    private void auditUser(@NotNull User user, @NotNull ISnowflake guild,
-            @NotNull IReplyCallback event) {
-        List<ActionRecord> actions =
-                actionsStore.getActionsByTargetAscending(guild.getIdLong(), user.getIdLong());
+    private Message auditMessage(long guildId, long targetId, int pageNo, @NotNull JDA jda) {
```

Comment:
Please dont abbreviate. `pageNumber`, not `pageNo`. Is it 0 or 1 based? What is the behavior if the number is invalid (negative, too high, ...)? I think this deserves a quick javadoc:
```java
/**
  * @param pageNumber blablabla
  */
```

Reply:
> What is the behavior if the number is invalid (negative, too high, ...)

i don't this that can ever be a case

Reply:
Not in your specific case. This is about making the code easy to understand for readers though. And to forge robust software, robust with respect to future modifications.

Someone might for example misinterpret the code and use `1` for the first page, where maybe the first page is `0`. Or might think that setting the page to 1000 would make it just use the highest page instead.

Please just make the behavior and intention clear, cause your code is not self-explanatory.

Reply:
```java
        // Handles the case of too low page number and too high page number
        pageNumber = Math.max(1, pageNumber);
        pageNumber = Math.min(totalPages, pageNumber);
```
so i should add this?

Reply:
> I think this deserves a quick javadoc:

```java
    /**
     * @param pageNumber page number to display when actions are divided into pages and each page can contain {@link AuditCommand#MAX_PAGE_LENGTH} actions
     */
```

this better?